autodiff/forward/real/real.hpp

```diff
@@ -797,13 +802,13 @@ constexpr auto min(const Real<N, T>& x, const Real<N, T>& y)
 template<size_t N, typename T, typename U, EnableIf<isArithmetic<U>>...>
 constexpr auto min(const Real<N, T>& x, const U& y)
 {
-    return (x[0] <= y) ? x : y;
+    return (x[0] <= y) ? x[0] : y;
```

Comment:
Your change will result in a _different return type_. Previously `min` returned `Real<N, T>` and now some mix of `T` and `U`, e.g. if both are of type `double` then it will just return a `double` value! Thus, you truncate all the derivatives.

Reply:
Feel free to have a look at my solution which leaves the interface unchanged. Maybe there is a better way? 

With the added T() operator, the method would not compile as it was previously. By the way, the same error happens if AUTODIFF_ENABLE_IMPLICIT_CONVERSION_REAL is defined.

Reply:
Your change to `return (x[0] <= y) ? x : Real<N, T>(y);` seems useful! Thanks!